### PR TITLE
ui: simplify/shorten error messages

### LINKED ISSUE #3706: run: ui issue for `run`
1. Eg:
```sh
$ dvc run -n train ...
ERROR: failed to run command - Stage with name 'train' already exists in 'pipelines.yaml'.
```
There's no need for `failed to run command` because the commands were never run and is a bit unnecessary.

2. 
```sh
$ dvc run -n train1 -d users.csv -p tr_conf.yaml:epochs,log_file,dropout -M summary.json -m logs.csv -o logs python train.py
ERROR: failed to run command - file/directory 'summary.json' is specified as an output in more than one stage:
	pipelines.yaml:train
This is not allowed. Consider using a different output name.
```

Suggestion:
```
ERROR: output 'summary.json' already exist on stage 'pipelines.yaml:train'.
```

3. 
```sh
$ dvc repro pipelines.yaml
ERROR: you've introduced a cycle in your pipeline that involves the following stages:
	- pipelines.yaml:train
	- pipelines.yaml:process
```

Suggestion:
```sh
ERROR: output 'IN.TSV' introduces a cycle that involves stages: train, process
```

### LINKED ISSUE #3718: ERROR: unexpected error - are any errors expected by the user?
The following example is to drive my point home, but, yeah, the `unexpected error` is unnecessary.
```sh
$ cd  `mktemp -d`
$ mkdir .dvc
$ dvc status
ERROR: unexpected error - expected str, bytes or os.PathLike object, not NoneType

Having any troubles? Hit us up at https://dvc.org/support, we are always happy to help!
```

### PR DESCRIPTION
**Pull request by skshetry**
* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here. If the CLI API is changed, I have updated [tab completion scripts](https://github.com/iterative/dvc/tree/master/scripts/completion).

* [x] ❌ I will check DeepSource, CodeClimate, and other sanity checks below. (We consider them recommendatory and don't expect everything to be addressed. Please fix things that actually improve code or fix bugs.)

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

closes #3706 and closes #3718

### CODE REVIEW

File: dvc/main.py [hunk 1]
```diff
@@ -58,6 +58,9 @@ def main(argv=None):
         ret = 253
     except DvcParserError:
         ret = 254
+    except DvcException:
+        ret = 255
+        logger.exception("")
```
Comment (skshetry) [inline dvc/main.py:63]: DvcException won't have `unexpected error` prepended. Other exceptions and assertion errors will still have that prefixed, which is a good compromise, I'd say.

File: dvc/stage/exceptions.py [hunk 2]
```diff
@@ -2,8 +2,12 @@
 
 
 class StageCmdFailedError(DvcException):
-    def __init__(self, stage):
-        msg = "{} cmd '{}' failed".format(stage, stage.cmd)
+    def __init__(self, stage, status=None):
+        period = "."
```
Comment (efiop) [inline dvc/stage/exceptions.py:6]: What's the point? :slightly_smiling_face:

File: dvc/exceptions.py [hunk 3]
```diff
@@ -23,13 +28,15 @@ class OutputDuplicationError(DvcException):
...
         assert all(hasattr(stage, "relpath") for stage in stages)
-        msg = (
-            "file/directory '{}' is specified as an output in more than one "
-            "stage: \n{}\n"
-            "This is not allowed. Consider using a different output name."
-        ).format(
-            output, "\n".join("\t{}".format(s.addressing) for s in stages)
-        )
+        if len(stages) == 1:
+            msg = "output '{}' is already specified on {}.".format(
```
Comment (efiop) [inline dvc/exceptions.py:32]: Maybe `in` instead of `on`?

Comment (efiop) [reply dvc/exceptions.py:32]: Also stage name should be wrapped in `'` to mark that it is generated. Same In messages below.

Comment (skshetry) [reply dvc/exceptions.py:32]: we have `stage: 'stage name'` as `__str__`. So, no need to wrap here. :smile:

**Pull request merged**